R/preproc.R

```diff
@@ -0,0 +1,149 @@
+#' @title Simple Pre-processing
+#'
+#' @description
+#' Function that offers a simple and direct way to train or predict [`PipeOp`]s and [`Graph`]s on [`Task`][mlr3::Task]s,
+#' [`data.frame`]s or [`data.table`][data.table::data.table]s.
+#'
+#' Training happens if `predict` is set to `FALSE` and no `state` is passed to this function.
+#' Prediction happens if `predict` is set to `TRUE` and if the passed `Graph` or `PipeOp` is either trained or a `state`
+#' is explicitly passed to this function.
+#'
+#' The passed  `PipeOp` or `Graph` gets modified-by-reference.
+#'
+#' @param indata ([`Task`][mlr3::Task] | [`data.frame`] | [`data.table`][data.table::data.table] )\cr
+#'   Data to be pre-processed.
+#' @param processor ([`Graph`] | [`PipeOp`])\cr
+#'   `Graph` or `PipeOp` which has exactly one input channel accepting a [`Task`][mlr3::Task] and one output channel.\cr
+#'   If `indata` is a [`data.frame`] or [`data.table`][data.table::data.table], the output channel must give a `Task` to be converted back into a
+#'   `data.frame` or `data.table`. Additionally, `processor`s which only work on sub-classes of
+#'   [`TaskSupervised`][mlr3::TaskSupervised] will not be accepted, as it would be unclear which column was the `target`.\cr
```

Comment:
```suggestion
#'   [`TaskSupervised`][mlr3::TaskSupervised] will not accept [`data.frame`] or [`data.table`][data.table::data.table], as it would be unclear which column was the `target`.\cr
```